### PR TITLE
Add skewed subtype for extrusion groups. Update the textwindow radio …

### LINKED ISSUE #1495: Add a checkbox to "allow skew" of extrusions
### System information

- **SolveSpace version:** ALL
- **Operating system:** ALL

### Expected behavior

It should be easy to make skewed extrusions. Perhaps a checkbox in the group text window to allow it.

### Actual behavior

Extrusions groups have 2 hidden perpendicular constraints to lock the extrusion direction perpendicular to the workplane. 

It turns out you can select a workplane different from the one containing a sketch to extrude the sketch, and this will create a skewed extrusion. This is a little awkward because the direction has to be set in the prior group via a little normal vector rather than using constraints in the extrude group.

### Additional information

I have tried commenting out the 2 perpendicular constraint, and it does result in a 3 DoF extrusion. All we'd really need is a checkbox for extrude groups to "allow skew" or similar which would inhibit creation of the 2 hidden constraint. This might be similar to how a constraint can be enabled via checkbox to lock helix pitch.

Is this a good idea?

### PR DESCRIPTION
**Pull request by phkahler**
…buttons and equation generation.

Closes #1495

### CODE REVIEW

File: src/sketch.h [hunk 1]
```diff
@@ -204,7 +204,8 @@ class Group {
         //WORKPLANE_BY_FACE          = 6004,
         // For extrudes, translates, and rotates
         ONE_SIDED                  = 7000,
-        TWO_SIDED                  = 7001
+        TWO_SIDED                  = 7001,
+        ONE_SKEWED                 = 7004
```
Comment (ruevs) [inline src/sketch.h]: Very cunning of you to introduce a new `Group::Subtype` in this way the saving is all taken care of. I like it :-)

But `TWO_SKEWED` is definitely required (as you already suggested).

File: src/textscreens.cpp [hunk 2]
```diff
@@ -377,14 +378,30 @@ void TextWindow::ShowGroupInfo() {
...
 
         bool one = (g->subtype == Group::Subtype::ONE_SIDED);
+        bool two = (g->subtype == Group::Subtype::TWO_SIDED);
+        bool skew = (g->subtype == Group::Subtype::ONE_SKEWED);
+        
+        if (g->type == Group::Type::EXTRUDE) {
+        Printf(false,
+            "%Ba   %f%Ls%Fd%s one-sided%E  "
+                  "%f%LS%Fd%s two-sided%E  "
+                  "%f%Lw%Fd%s skewed%E",
```
Comment (ruevs) [inline src/textscreens.cpp]: I think it is better if this is called "perpendicular" and checked by default.

**Review by ruevs — CHANGES_REQUESTED** [review]
I made an extra commit in this branch with my proposed changes.

File: src/textscreens.cpp (hunk 2 above)
Comment (phkahler) [reply src/textscreens.cpp]: > I think it is better if this is called "perpendicular" and checked by default.

Disagree. I think the checkbox should be "skewed" and unchecked by default. The normal/expected/natural behavior should not need to be enabled (even by default) and I kind of dislike the word "prependicular" vs "skewed". Maybe it's too many syllables...  ;-)

**Comment by phkahler** [comment]
You're the best @ruevs!

**Pull request merged**